docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,540 @@
+```yaml
+type: MeshTrafficPermission
+name: global
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        action: DENY
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          kuma.io/zone: us-east
+      default:
+        action: ALLOW
+---
+type: MeshTrafficPermission
+name: backend
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  from:
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          env: dev
+      default:
+        action: DENY
+```
+
+Concatenated "from" array
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    default:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    default:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        env: dev
+    default:
+      action: DENY
+```
+
+2. Create a full rule-based view (with negations) 
+
+Regular [rule-based view](005-policy-matching.md#rule-based-view) doesn't work well for NetworkRBAC generation,
+because rule-based view relies on the order, but NetworkRBAC doesn't. 
+
+In this example, if a request matches "rule 2" this means it doesn't match "rule 1":
+```yaml
+rules:
+  - targetRef: # rule 1
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    default:
+      action: DENY
+  - targetRef: # rule 2
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    default:
+      action: ALLOW
+```
+
+So "rule 2" implicitly matches requests from zone "us-east" that are **not** from env "dev".
+That's why it's important to have a full rule-based view with negations ("!" means "not"):
+
+```yaml
+rules:
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    default:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: !dev
+    default:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: dev
+    default:
+      action: DENY 
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: !dev
+    default:
+      action: DENY
+```
+
+See [how to generate a full rule-based view with negations](#how-to-generate-a-full-rule-based-view-with-negations)
+
+3. For each "rule" that has "action: ALLOW" generate a single "principal" in v3.RBAC:
+
+```yaml
+action: ALLOW
+policies:
+  "backend": # name of the most specific MeshTrafficPermission policy for the DPP
+    permissions:
+      - any: true
+    principals: # each principal is matched with OR semantics
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+```
```

Comment:
Didn't we talked about a policy with "or_ids" at the top level?

Reply:
I missed this fact before that 1 policy has a **list** of principals, and principals are matched with OR semantic. So I added `For each "rule" that has "action: ALLOW" generate a single "principal" in v3.RBAC`.

Reply:
Could this example be expand to something less trivial then? We're trying to capture in a MADR the decisions and the reasons we're taking these decisions.

Reply:
slightly improved the example to have at least 2 "allow" rule at this step